### PR TITLE
Add test to verify generated files

### PR DESCRIPTION
**Pull request by csrwng**
Also updates generated out-of-date deepcopy file.

### CODE REVIEW

**Comment by csrwng** [comment]
@joelddiaz @twiest this should be ready for review now

File: hack/verify-generated.sh [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+	echo "Generated deepcopy file is different. Run 'make generate'"
+	exit 1
+fi
+if ! diff -Naup "./pkg/operator/assets/bindata.go" "${tempdir}/bindata.go"; then
+	echo "Generated bindata file is different. Run 'make generate'"
+	exit 1
+fi
+
+mkdir "${tempdir}/crds"
+cp ${SRC_DIR}/config/crds/* "${tempdir}/crds/"
```
Comment (twiest) [inline hack/verify-generated.sh]: Would it make sense to move this up before any of the generations?

I'm know that today the things in the `crds` directory are only modified by `make manifests`, but what if in the future something in the `crds` directory gets modified by `make generate` ?

In other words, why not copy all of the source compares before making any changes?

Comment (csrwng) [reply hack/verify-generated.sh]: ack, makes sense

Comment (csrwng) [reply hack/verify-generated.sh]: thx @twiest, code updated

**Pull request merged**